### PR TITLE
Moar test coverage

### PR DESCRIPTION
**Pull request by kyle-laplante**
# Citrine Python PR

## Description 
Please briefly explain the goal of the changes/this PR.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.

### CODE REVIEW

File: src/citrine/resources/data_concepts.py [hunk 1]
```diff
@@ -140,8 +140,6 @@ def _get_field(data, field: str):
             return data.get(field)
         elif isinstance(data, object):
             return getattr(data, field, None)
-        else:
-            TypeError("Expected data to be a dictionary or object, instead got {}".format(data))
```
Comment (kyle-laplante) [inline src/citrine/resources/data_concepts.py]: logically unreachable...the line before tests for object....everything is an object

File: tests/informatics/test_scorers.py [hunk 2]
```diff
@@ -37,6 +37,14 @@ def mei_scorer() -> MEIScorer:
     )
 
 
+def test_mli_repr(mli_scorer):
+    assert str(mli_scorer) == "<MLIScorer 'MLI(z)'>"
```
Comment (jtharris) [inline tests/informatics/test_scorers.py]: You can add these to the rest of the `str` checks in `test_resources.py`

File: tests/informatics/test_processors.py [hunk 3]
```diff
@@ -16,6 +16,14 @@ def enumerated_processor() -> EnumeratedProcessor:
     return EnumeratedProcessor('my enumerated thing', 'enumerates the things', 10)
 
 
+def test_grid_repr(grid_processor):
+    assert str(grid_processor) == "<GridProcessor 'my thing'>"
```
Comment (jtharris) [inline tests/informatics/test_processors.py]: You can add these to the rest of the `str` checks in test_resources.py

**Comment by kyle-laplante** [comment]
> Some organization notes - we can collapse all those `str` checks where the others are.

oh yah for sure

**Pull request merged**